### PR TITLE
added gauss_seidal method

### PR DESCRIPTION
**Pull request by AyushTripathiIITP**
### Describe your change:

Added Gauss Seidal method for solving equations

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: arithmetic_analysis/gauss_seidal.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+def seidel(a, x, b):
```
Comment (cclauss) [inline arithmetic_analysis/gauss_seidal.py:1]: Please change the filename to have to proper spelling and add this URL:|
* https://en.wikipedia.org/wiki/Gauss%E2%80%93Seidel_method

Comment (AyushTripathiIITP) [reply arithmetic_analysis/gauss_seidal.py:1]: @cclauss can you please help me with this error further I am new to this setup and know little why this error is occuring

Comment (cclauss) [reply arithmetic_analysis/gauss_seidal.py:1]: Just below there is a hotlink `Ruff (PIE808)`.  Please click that and read the rule that our linter `Ruff` is complaining about.  See if you can make a change that will satisfy it.  There are three `PIE808` violations in this code that you will need to fix.  Repeat this for the other Ruff annotations below.

File: arithmetic_analysis/gauss_seidal.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    a = []
+    b = []
+    # initial solution depending on n(here n=3)
+    x = [0, 0, 0]
+    a = [[4, 3, 2], [3, 5, 1], [1, 1, 3]]
+    b = [4, 7, 3]
+    print(x)
+
+    # loop run for m times depending on m the error value
+    for i in range(0, 25):
```
Comment (cclauss) [inline arithmetic_analysis/gauss_seidal.py:31]:
```suggestion
    for _ in range(25):
```

**Comment by tianyizheng02** [comment]
Plagiarized from https://www.geeksforgeeks.org/gauss-seidel-method/

**Pull request closed without merging**